### PR TITLE
Windows packaging build pipeline for C-api packages (CPU and GPU)

### PR DESCRIPTION
**Pull request by shahasad**

### CODE REVIEW

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -493,13 +493,15 @@ def run_onnx_tests(build_dir, configs, onnx_test_data_dir, provider, enable_para
           cmd.append(model_dir)
         if os.path.exists(onnx_test_data_dir):
           cmd.append(onnx_test_data_dir)
-        run_subprocess([exe] + cmd, cwd=cwd)
```
Comment (snnn) [inline tools/ci_build/build.py]: why do you remove this?

Comment (shahasad) [reply tools/ci_build/build.py]: I moved it to line 503. Looks like we were running the same test twice when parallel test is requested. Was there any reason for that?

Comment (snnn) [reply tools/ci_build/build.py]: The arguments are different. 
If enable_parallel_executor_test is true, we need to run the tests twice:
1. with parallel executor enabled ('-x')
2. without parallel executor ( without '-x')

And, your changes  disabled multiple threaded test(run multiple inferences at the same time), this is the biggest concern I have.

Comment (shahasad) [reply tools/ci_build/build.py]: sent an update.

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -551,7 +554,7 @@ def main():
         cmake_extra_args = []
         if(is_windows()):
           if (args.x86):
-            cmake_extra_args = ['-A','Win32','-G', 'Visual Studio 15 2017']
+            cmake_extra_args = ['-A','Win32','-T','host=x64','-G', 'Visual Studio 15 2017']
```
Comment (snnn) [inline tools/ci_build/build.py]: Why setting toolchain to x64 even for x86 build?

File: tools/ci_build/build.py [hunk 3]
```diff
@@ -493,13 +493,16 @@ def run_onnx_tests(build_dir, configs, onnx_test_data_dir, provider, enable_para
...
+        
         if enable_parallel_executor_test:
+          run_subprocess([exe] + cmd, cwd=cwd)  
           if provider == 'mkldnn':
             #limit concurrency to 1
             run_subprocess([exe,'-x', '-c', '1', '-j', '1'] + cmd, cwd=cwd)
           else:
             run_subprocess([exe,'-x'] + cmd, cwd=cwd)
+        else:
+          run_subprocess([exe, '-c', '1', '-j', '1'] + cmd, cwd=cwd)
```
Comment (snnn) [inline tools/ci_build/build.py]: Why adding '-c 1 -j 1'? if it's needed for x86 build, can we restrict it only for x86 ?

File: tools/ci_build/build.py [hunk 4]
```diff
@@ -487,8 +487,10 @@ def run_onnx_tests(build_dir, configs, onnx_test_data_dir, provider, enable_para
         cmd = []
         if provider:
           cmd += ["-e", provider]
-          if provider == 'cuda':
-            cmd += ["-j", "2"]
+
+        if num_parallel_models > 0:
+          cmd += ["-j", str(num_parallel_models)]
```
Comment (snnn) [inline tools/ci_build/build.py]: So, we won't test the builtin cpu provider with num_parallel_models>0?

File: tools/ci_build/build.py (hunk 2 above)
Comment (shahasad) [reply tools/ci_build/build.py]: it goes out of heap space with 32 bit toolchain

File: tools/ci_build/build.py (hunk 4 above)
Comment (shahasad) [reply tools/ci_build/build.py]: plz check the parameters set on line 600 onwards, I believe, the execution flow remains the same for the 3 cases that sets num_parallel_models = 1

File: onnxruntime/test/onnx/main.cc [hunk 5]
```diff
@@ -354,6 +354,28 @@ int real_main(int argc, char* argv[]) {
   broken_tests["fp16_inception_v1"] = "need to adjust the per_sample_tolerance: 0.002";
 #endif
 
+#ifdef _WIN32
+  broken_tests["resnet50"] = "failed: type mismatch";
```
Comment (shahasad) [inline onnxruntime/test/onnx/main.cc]: These tests currently fail in the pipeline for x86 build. However, if run the test on individual model, each test pass. So, they are more of problem in the test framework. Need to be fixed (TODO)

```
PS C:\Users\prs\work_projects\ort_x86> .\build\Windows\RelWithDebInfo\RelWithDebInfo\onnx_test_runner -j 1 -n resnet101v2 -n resnet50 -n tf_resnet_v1_50 -n tf_resnet_v2_50 -n tf_resnet_v1_101 -n tf_resnet_v1_152 -n tf_resnet_v2_101 -n tf_resnet_v2_152 -n vgg19 -n tf_nasnet_large -n tf_pnasnet_large -n resnet101v2 -n resnet152v2 -n tf_inception_resnet_v2 -n tf_inception_v4 -n tf_inception_v3 -n resnet50v2 C:\Users\prs\Downloads\20190130\ C:\Users\prs\work_projects\onnxruntime\cmake\external\onnx\onnx\backend\test\data

2019-03-05 17:49:46.7401428 [W:onnxruntime:Default, graph.cc:2193 onnxruntime::Graph::CleanUnusedInitializers] gpu_0/imagenet1k_blobs_queue_f22e83c9-22cd-4a8b-a66d-113af6b832b4_0 exists in this graph's initializers but it is not used by any node

2019-03-05 17:49:46.8097771 [W:onnxruntime:resnet50, session_state_initializer.cc:499 onnxruntime::SaveInputOutputNamesToNodeMapping] Graph input with name gpu_0/imagenet1k_blobs_queue_f22e83c9-22cd-4a8b-a66d-113af6b832b4_0 is not associated with a node.

2019-03-05 17:50:22.5466323 [W:onnxruntime:Default, graph.cc:2193 onnxruntime::Graph::CleanUnusedInitializers] gpu_0/imagenet1k_blobs_queue_f22e83c9-22cd-4a8b-a66d-113af6b832b4_0 exists in this graph's initializers but it is not used by any node

2019-03-05 17:50:22.6130785 [W:onnxruntime:resnet50, session_state_initializer.cc:499 onnxruntime::SaveInputOutputNamesToNodeMapping] Graph input with name gpu_0/imagenet1k_blobs_queue_f22e83c9-22cd-4a8b-a66d-113af6b832b4_0 is not associated with a node.

result:

        Models: 40

        Total test cases: 363

                Succeeded: 363

                Not implemented: 0

                Failed: 0

        Stats by Operator type:

                Not implemented(0):

                Failed:

Failed Test Cases:
```

Comment (shahasad) [reply onnxruntime/test/onnx/main.cc]: @pranavsharma

Comment (snnn) [reply onnxruntime/test/onnx/main.cc]: It worth to investigate.

**Pull request merged**